modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/OverviewsController.java

```diff
@@ -89,41 +90,72 @@ int pickOverviewLevel(final OverviewPolicy policy, final double[] requestedResol
         // requested scale factor for least reduced axis
         final OverviewLevel max = (OverviewLevel) resolutionsLevels.get(0);
 
+        // the requested resolutions (even virtual)
+        boolean useVirtual = false;
+        if (virtualNativeResolution != null){
+            useVirtual = true;
+        }
+
         // the requested resolutions
-        final double requestedScaleFactorX;
-        final double requestedScaleFactorY;
+        double requestedScaleFactorX;
+        double requestedScaleFactorY;
+        double virtualRequestedScaleFactorX;
+        double virtualRequestedScaleFactorY;
+        double virtualRequestedScaleFactor = Double.NaN;
+        double reqx = 0;
+        double reqy = 0;
         if (requestedResolution != null) {
-            final double reqx = requestedResolution[0];
-            final double reqy = requestedResolution[1];
-            requestedScaleFactorX = reqx / max.resolutionX;
-            requestedScaleFactorY = reqy / max.resolutionY;
-        } else {
+            reqx = requestedResolution[0];
+            reqy = requestedResolution[1];
+            if (useVirtual) {
+                virtualRequestedScaleFactorX = virtualNativeResolution[0] / max.resolutionX;
+                virtualRequestedScaleFactorY = virtualNativeResolution[1] / max.resolutionY;
+                virtualRequestedScaleFactor = (virtualRequestedScaleFactorX <= virtualRequestedScaleFactorY) ? 
+                        virtualRequestedScaleFactorX : virtualRequestedScaleFactorY;
+
+            }
+        }
+        if (!useVirtual && requestedResolution == null){

```

Comment:
I'd add a comment as to why we exit, or not, here

Reply:
The old code was returning 0 (first image, highres) as soon as requestedResolution is null.
The new code should check if a different imageIndex need to be returned when virtualResolution has been specified so it need to perform some additional checks.

I'm adding comment to the code too.